enhancements/csvless-bundles.md

```diff
@@ -24,209 +24,488 @@ status: implementable
 
-This intermediary step (creating a CSV internally to represent the kubernetes objects) is a delibrate decision in 4.5 to help focus and deliver the feature. This may be relaxed in future release versions.
+_Ex._
 
-#### Validation
+```sh
+$ opm bundle resources . # detect the mediatype of the current directory, generate the resources to apply for it as OLM would, and write the YAML stream to stdout
+$ opm bundle resources -o my-bundle.yaml image://quay.io/my/bundle:v1.0.0 # pull and unpack the bundle image, generate the resources to apply for it as OLM would, and write the YAML stream to a file
+```
 
-The [operator-framework static validation package](https://github.com/operator-framework/api/tree/master/pkg/validation) can be extended and used to validate the content being fed into a `registry+v2` bundle build. New validators will be added to `opm` which ensure all required manifests and metadata are present and valid.
+#### Validate Bundles
 
-#### Indexes
+As a Pipeline Maintainer, I want a way to validate `registry+v2` bundles, so that I can ensure OLM is able to unpack and apply it before I add it to any release index.
 
-When a `registry+v2` bundle is added to an index, `opm` will also synthesize and add a CSV. A lot of the existing `opm` and `operator-registry` code assumes bundles always contain CSVs, which make synthesis the simplest path to supporting `registry+v2` bundles -- in the future, we may want to revisit this decision, especially when the [`Operator` API](https://github.com/openshift/enhancements/blob/master/enhancements/olm/simplify-apis.md#proposal) is released.
+##### UX
 
-To fill out the dependency graph between versions of bundles, OLM considers the "replaces" field on the CSV which points to the previous version of the operator. This field will need to be supplied in the additional metadata the operator author provides in the bundle so that OLM can build dependency graphs. There is another enhancement around semantic versioning that would remove this requirement altogether, but it is independent of this enhancement.
+Add a new command to `opm`, `opm bundle validate`, which takes as input an Operator Bundle and fails with informative errors should the bundle be invalid.
+
+```sh
+$ opm bundle validate . # validate the bundle in the current directory, printing the results to stdout
+$ opm bundle validate image://quay.io/my/bundle:v1.0.0 -o results.xml # pull, unpack, and validate the remote bundle image, printing the results to a file
+```
 
-Supporting indexes with different mediatypes is a key requirement. Both `registry+v2`  and `registry+v1` type bundles should be able to be added to an existing index. There is the potential of versioning the index itself to help differentiate between newer index images, that include the `registry+v2` mediatype, and older versions.
+Results are in [XUnit format](todo).
```

Comment:
I'm not convinced this is the best fit for validation results. Could you sketch out what this would look like, or maybe make a more general statement about how the validation results can be consumed? Specifics are probably outside the scope of this document.

Reply:
> I'm not convinced this is the best fit for validation results.

Yeah, given recent discussions on this topic, I'm inclined to agree.

> Could you sketch out what this would look like, or maybe make a more general statement about how the validation results can be consumed? 

Maybe a binary output would be better and we could leave it to individual "validation rules" to handle their own additional output. Can I both mention that we need to validate the bundle format and side-step the validation framework structure?

Reply:
> Can I both mention that we need to validate the bundle format and side-step the validation framework structure?

SGTM. That's its own can of worms.